hack/tools/catalogs/lib/bundle.sh

```diff
@@ -0,0 +1,191 @@
+# Library of functions for interacting with bundles and their manifests
+
+# SCRIPT_ROOT is the root of the script
+source "$(dirname "${BASH_SOURCE[0]}")/hash.sh"
+
+# Given package and version grabs the bundle image from stdin FBC stream
+function get-bundle-image(){
+    local package_name="${1}"
+    local package_version="${2}"
+    local image
+    image=$(jq -r --arg pkg "$package_name" --arg ver "$package_version" \
+        'select(.schema == "olm.bundle" and (.properties[] | select(.type == "olm.package" and .value.packageName == $pkg and .value.version == $ver))) | .image')
+
+    if [ -z "$image" ]; then
+        echo "ERROR: No matching image found for package '$package_name' with version '$package_version'." >&2
+        exit 1
+    fi
+
+    echo "${image}"
+}
+
+function is-all-namespace-mode-enabled() {
+    local csv="${1}"
+    local valid
+    valid=$(yq eval '(.spec.installModes[] | select(.type == "AllNamespaces" and .supported == true)) and true' "${csv}")
```

Comment:
Document that `yq` is required where this is used?

Reply:
Out of curiosity, why does this need the additional `and true` at the end?

Reply:
I put it there to force yq to evaluate it to "true" (or false) rather than returning the matching snippet (or nothing). E.g. without it, it would look something like this:

```
match=$(yq eval '(.spec.installModes[] | select(.type == "AllNamespaces" and .supported == true))' "${csv}")
    if [ -n "${match}" ]; then
        echo "true"
    else
        echo "false"
    fi
```

I'll add a comment explaining it